src/package-linker.js

```diff
@@ -566,7 +566,7 @@ export default class PackageLinker {
       const ref = pkg._reference;
       invariant(ref, 'Package reference is missing');
       // TODO: We are taking the "shortest" ref tree but there may be multiple ref trees with the same length
-      const refTree = ref.requests.map(req => req.parentNames).sort((arr1, arr2) => arr1.length - arr2.length)[0];
```

Comment:
I don't think this is correct. The sort is there to have a consistent iteration order across yarn runs. If it works without the sorting, it doesn't mean that the bug is fixed, just that it doesn't appear on whatever random order the data have been layout.

Reply:
yes, you're correct that `[].every(() => false) === true`. However the issue is not with `req.parentNames`. The`[<value>].every((name, idx) => name === refTree[idx])

Reply:
Am I correct that the goal here is to skip refs with parentNames.length === 0? If so you should just filter those ones out and then sort.